### PR TITLE
JUCX add checkstyle, licence and header generation plugin.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What

1. Added checkstyle rules, based on [Google java style guide](https://google.github.io/styleguide/javaguide.html), with small modifications (4 spaces).
2. Added licence into pom.xml, so we could publish library on [Maven central](https://maven.org/).
3. Added native-maven-plugin, to automatically generate jni header file from Bridge.java class (command `mvn native:javah`).
4. Minor fixes to pass checkstyle.

## Why ?
Prepare common stuff for JUCX development. TODO: add to jenkins tests JUCX tests.

### CODE REVIEW

**Comment by shamisp** [comment]
IMHO This are basic styling fixed that can be applied right away.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8708/ for details (Mellanox internal link).

File: bindings/java/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+<!--
+    Checkstyle configuration that checks the Google coding conventions from Google Java Style
+    that can be found at https://google.github.io/styleguide/javaguide.html.
+
+    Checkstyle is very configurable. Be sure to read the documentation at
+    http://checkstyle.sf.net (or in your downloaded distribution).
+
+    To completely disable a check, just comment it out or delete it from the file.
+
+    Authors: Max Vetrenko, Ruslan Diachenko, Roman Ivanov.
```
Comment (yosefe) [inline bindings/java/checkstyle.xml]: does it have a license?

Comment (petro-rudenko) [reply bindings/java/checkstyle.xml]: This file is just configuration for checkstyle copied and modified from here:(https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml).
We can add project licence here, for example like in apache: https://github.com/apache/kudu/blob/148a0c7bec6554724339a2235cbd723fb74be339/java/kudu_style.xml#L2

Comment (yosefe) [reply bindings/java/checkstyle.xml]: adding @shamisp 
checkstyle seems to be LGPL, while UCX is BSD-3, so it's not compatible
not sure if it's ok that Apache copied it
we need to check this with legal

Comment (petro-rudenko) [reply bindings/java/checkstyle.xml]: Maven checkstyle plugin that we use is an apache licenced: https://github.com/apache/maven-checkstyle-plugin
This file is just a bunch of configurations. And we don't include it into binary - it's only at compile time.

Comment (petro-rudenko) [reply bindings/java/checkstyle.xml]: Spark uses the same: https://github.com/apache/spark/blob/2977e2312d9690c9ced3c86b0ce937819e957775/dev/checkstyle.xml - they just licenced it under Apache

Comment (shamisp) [reply bindings/java/checkstyle.xml]: AFAIK LGPL is not compatible with BSD3. @yosefe can you please check with legal ?

Comment (shamisp) [reply bindings/java/checkstyle.xml]: https://opensource.stackexchange.com/questions/6698/can-i-use-code-from-an-lgpl-v2-licenced-project-in-a-bsd-3-clause-licenced-proje

Comment (petro-rudenko) [reply bindings/java/checkstyle.xml]: The same situation as for example with this file: https://github.com/openucx/ucx/blob/master/debian/rules.in#L4

Original file from debian repo is GNU licensed. It's just a config file.

Comment (shamisp) [reply bindings/java/checkstyle.xml]: @yosefe did we approve this one ?
I see it has special exception:
```As a special exception, when this file is copied by dh-make into a
# dh-make output file, you may use that output file without restriction.```

Comment (petro-rudenko) [reply bindings/java/checkstyle.xml]: Here's google itself using this file in it's open source projects: https://github.com/google/graphicsfuzz/blob/066b518b8f16be2853d9927116d0d6f85c88c2ca/checkstyle-config/src/main/resources/graphicsfuzz/checkstyle.xml
See this config in millions of projects under different licenses. 

Again this is used only at compile time. Like here used other GNU tools ( make, libtool, etc.)

Comment (petro-rudenko) [reply bindings/java/checkstyle.xml]: As discussed, added checkstyle with Apache header + Apache Licence

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8752/ for details (Mellanox internal link).

**Pull request merged**